scripts/docs/features-html-output/build-features-html-output.js

```diff
@@ -0,0 +1,571 @@
+ * @returns {Boolean}
+ */
+function isClosingTag( line ) {
+	return line.startsWith( '</' );
+}
+
+/**
+ * Checks, if there is a closing tag for currently examined opening tag.
+ *
+ * @param {String} tag Currently examined opening tag.
+ * @param {Array.<String>} lines Next lines to search for a closing tag.
+ * @returns {Boolean}
+ */
+function hasClosingTagFor( tag, lines ) {
+	const closingTag = tag.replace( /<([a-z]+).*>/, '</$1>' );
+
+	return lines.some( line => line === closingTag );
+}
+
+/**
+ * Indents a line by a specified number of characters.
+ *
+ * @param {String} line Line to indent.
+ * @param {Number} indentCount Number of characters to use for indentation.
+ * @param {String} [indentChar] Indentation character.
+ * @returns {String}
+ */
+function indentLine( line, indentCount, indentChar = '\t' ) {
+	return `${ indentChar.repeat( indentCount ) }${ line }`;
+}
+
+/**
+ * Helper, which transforms any value to an array. If the provided value is already an array, it is returned unchanged.
+ *
+ * @param {*} data The value to transform to an array.
+ * @returns {Array.<*>} An array created from data.
+ */
+function toArray( data ) {
+	return Array.isArray( data ) ? data : [ data ];
+}
+
+/**
+ * Helper (factory), which creates a function, that prepends and/or appends provided value by another value.
+ *
+ * @param {Object} options Options to define prefix and/or suffix.
+ * @param {String} [options.prefix] A string to add as a prefix to provided value. Empty string by default.
+ * @param {String} [options.suffix] A string to add as a suffix to provided value. Empty string by default.
+ * @returns {Function}
+ */
+function wrapBy( { prefix = '', suffix = '' } = {} ) {
+	return item => `${ prefix }${ item }${ suffix }`;
+}
+
+/**
+ * @typedef {Object} HtmlOutput
+ * @property {String|Array.<String>} elements HTML elements, that are created or altered by the plugin.
+ * @property {String|Array.<String>} classes CSS class names, that may be applied to the HTML elements.
+ * @property {String|Array.<String>} styles Inline CSS styles, that may be applied to the HTML elements.
+ * @property {String|Array.<String>} attributes Other HTML attributes, that may be applied to the HTML elements.
+ * @property {String} implements A name of an element or a pseudo-element, which classes, styles or attributes may be inherited from.
+ * @property {String} _comment A human-readable description.
+ */
+
+/**
+ * @typedef {Object} Plugin
+ * @property {String} name Plugin name.
+ * @property {String} className Plugin class name.
+ * @property {String} docs An absolute or relative URL to the plugin's documentation.
+ * @property {String} path A path to the file, relative to the metadata file, that exports the plugin.
+ * @property {HtmlOutput} htmlOutput An array of objects, that defines all possible HTML elements which can be created by a given plugin.
+ */
+
+/**
+ * @typedef {Object} ParsedPlugin
+ * @property {String} name Plugin name.
```

Comment:
This actually isn't a plugin name. But for me, it's quite hard to define such property names better :sweat\_smile:

Reply:
But the `pluginNameMarkup` would be a bit better IMO. And I'd use `htmlOutputMarkup` as well as this is the markup representation of the plugins output.

Reply:
Done.